Keas.Mvc/ClientApp/components/Access/AccessAssignmentContainer.tsx

```diff
@@ -273,27 +183,84 @@ class AssignmentContainer extends React.Component<IProps, IState> {
       throw new Error(); // throw error so modal doesn't close
     }
 
-    const assignments = this.state.assignments;
+    const assignmentsData = assignments;
     assignments.push(accessAssignment);
-
-    this.setState({
-      assignments
-    });
-
-    this.props.onAssignSuccess(accessAssignment);
+    setAssignments(assignmentsData);
+    props.onAssignSuccess(accessAssignment);
   };
 
-  private _openDetails = (access: IAccess) => {
-    this.props.history.push(
-      `/${this.context.team.slug}/access/details/${access.id}`
-    );
+  const openDetails = (access: IAccess) => {
+    history.push(`/${context.team.slug}/access/details/${access.id}`);
   };
 
-  private _getBaseUrl = () => {
-    return this.props.person
-      ? `/${this.context.team.slug}/people/details/${this.props.person.id}`
-      : `/${this.context.team.slug}/access/details/${this.props.access.id}`;
+  const getBaseUrl = () => {
+    return props.person
+      ? `/${context.team.slug}/people/details/${props.person.id}`
+      : `/${context.team.slug}/access/details/${props.access.id}`;
   };
-}
 
-export default withRouter(AssignmentContainer);
+  if (!PermissionsUtil.canViewAccess(context.permissions)) {
+    return <Denied viewName='Access' />;
+  }
+  const { action, assetType } = params;
+  const isRevokeModalShown =
+    assetType === 'accessAssignment' && action === 'revoke';
+  const isEditModalShown =
+    assetType === 'accessAssignment' && action === 'update';
+  const isAssignModalShown = assetType === 'access' && action === 'assign';
+
+  return (
+    <div>
+      {isRevokeModalShown && (
+        <RevokeAccess
+          assignment={selectedAssignment}
+          revoke={callRevoke}
```

Comment:
Maybe as a future optimization but when you passalong a callback, like callRevoke or any of those similar methods, you should use the `useCallback` hook to prevent extra unnecessary network calls.  It might be worth running this with the network tab open to make sure it's not doing anything to weird with this code already.

Reply:
I think we can take a look at this in a future branch since we pass around multiple callbacks into components.